crates/sparrow-syntax/src/syntax/signature.rs

```diff
@@ -39,9 +88,23 @@ impl Signature {
             )
         }
 
+        // check that no duplicates exist in the type_parameters
```

Comment:
Maybe I missed it, is there a check for using a type variable in the return position only? 
e.g., 

```
 foo<I: number, T: any>(in: I): T
```

Reply:
On the same vein, a check for using an undefined type variable:

```
foo<I: number>(in: N): N
```